File: lib_common/src/main/java/com/fly/tour/common/view/LoadingTransView.java
package com.fly.tour.common.view;

import android.content.Context;
import android.graphics.drawable.AnimationDrawable;
import android.util.AttributeSet;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import com.fly.tour.common.R;

/**
 * Description: <LoadingView><br>
 * Author:      gxl<br>
 * Date:        2019/3/25<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
 */
public class LoadingTransView extends RelativeLayout {
    private final AnimationDrawable mAnimationDrawable;
    public LoadingTransView(Context context, AttributeSet attrs) {
        super(context, attrs);
        inflate(context, R.layout.view_trans_loading,this);
        ImageView imgLoading = findViewById(R.id.img_trans_loading);
        mAnimationDrawable = (AnimationDrawable) imgLoading.getDrawable();
    }
    public void startLoading(){
        mAnimationDrawable.start();
    }
    public void stopLoading(){
        mAnimationDrawable.stop();
    }
    public void loading(boolean b){
        if(b){
            startLoading();
        }else{
            stopLoading();
        }
    }
}

File: lib_common/src/main/java/com/fly/tour/common/view/LoadingTransView.kt
package com.fly.tour.common.view

import android.content.Context
import android.graphics.drawable.AnimationDrawable
import android.util.AttributeSet
import android.view.View
import android.widget.ImageView
import android.widget.RelativeLayout
import com.fly.tour.common.R

/**
 * Description: <LoadingView><br>
 * Author:      mxdl<br>
 * Date:        2019/3/25<br>
 * Version:     V1.0.0<br>
 * Update:     <br>
</LoadingView> */
class LoadingTransView(context: Context, attrs: AttributeSet) : RelativeLayout(context, attrs) {
    private val mAnimationDrawable: AnimationDrawable

    init {
        View.inflate(context, R.layout.view_trans_loading, this)
        val imgLoading = findViewById<ImageView>(R.id.img_trans_loading)
        mAnimationDrawable = imgLoading.drawable as AnimationDrawable
    }

    fun startLoading() {
        mAnimationDrawable.start()
    }

    fun stopLoading() {
        mAnimationDrawable.stop()
    }

    fun loading(b: Boolean) {
        if (b) {
            startLoading()
        } else {
            stopLoading()
        }
    }
}
